src/devices.jl

```diff
@@ -85,6 +85,15 @@ function device()
     return DeviceConstructor()
 end
 
+"""
+    device_summary(device)
+
+Return the device type. If using a CUDADevice, return device information and UUID.
+
+Internal function, used with `Base.summary(context)`
+"""
+device_summary(device) = device_type()
```

Comment:
```suggestion
device_summary(device) = string(device_type())
```
Should we convert this to a string so that the cpu and cuda versions both return strings?

Reply:
Also, I'm wondering if we need a separate function for this, or if we should just overload `Base.summary` for it.